File: backend/src/main/java/org/volcampanion/domain/Talk.java
package org.volcampanion.domain;

import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
public class Talk extends IdentifiableDomain {
    private String title;
    private String description;
    private String level;
    private String language;
    private TalkTheme theme;
    private TalkFormat format;
    private Conference conference;
    private List<Speaker> speakers;
}

File: frontend/src/Data/DTO/Talk.ts
import {Conference} from "./Conference";
import {Theme} from "./Theme";
import {Format} from "./Format";
import {Speaker} from "./Speaker";

export interface TalkTeaser {
  id: number;
  title: string;
  description: string;
  level: string;
  language: string;
  conference: Conference;
  format: Format;
}

export interface Talk extends TalkTeaser {
  theme: Theme;
  speakers: Speaker[];
}



